lib/active_model/serializer/adapter.rb

```diff
@@ -94,6 +98,38 @@ def include_meta(json)
         json[meta_key] = meta if meta
         json
       end
+
+      def key_format
+        serializer.key_format || default_key_format
+      end
+
+      def format_keys(formattable)
+        return formattable if !formattable.respond_to? :each_with_object ||
+                              key_format == :unaltered
+
+        if formattable.respond_to? :keys
+          formattable.each_with_object({}) do |(key, value), formatted_hash|
+            formatted_hash[format_key(key)] = format_keys(value)
+          end
+        elsif formattable.respond_to? :<<
+          formattable.each_with_object(formattable.class.new) do |item, shovelable|
+            shovelable << format_keys(item)
+          end
+        end
+      end
+
+      def format_key(formattable_key)
+        case key_format
```

Comment:
@timurvafin Do you have a ref for `Admin::User` -> `user`?  I'm not sure what I would do in such a case.